certbot/plugins/dns_common.py

```diff
@@ -34,8 +53,30 @@ def add_parser_arguments(cls, add, default_propagation_seconds=10):  # pylint: d
         add('propagation-seconds',
             default=default_propagation_seconds,
             type=int,
-            help='The number of seconds to wait for DNS to propagate before asking the ACME server '
-                 'to verify the DNS record.')
+            help='The number of seconds to wait for DNS to propagate before '
+                 'asking the ACME server to verify the DNS record.')
+        add('override-challenge',
+            action=OverrideChallengeAction,
+            default="{acme}",
+            help='Override default challenge for following domains on the '
+                 'command line. Validation depends on a CNAME having been '
+                 'provisioned at the standard location '
+                 '(_acme-challenge.<domain>), pointing to the alternate '
+                 'location. The specified value may include "{domain}" '
+                 'and "{acme}" format strings, which will respectively '
+                 'expand to the domain being requested, and the corresponding '
+                 'standard ACME challenge location.')
+
+    def validation_domain_name(self, achall):  # pylint: disable=missing-docstring
+
+        domain = achall.domain
+        acme_loc = achall.validation_domain_name(achall.domain)
+
+        challenge_map = self.conf('override-challenge-map')
```

Comment:
Ah, indeed an oversight.
Test coverage here will be interesting, as currently all DNS plugin tests provide a fake config which is assumed to reflect the actual namespace from argument parsing, and here what we'd want is a test based on the actual argument parser.

Reply:
Sorry, I should have been more clear here. Mocking or changing the test config object is completely fine for this purpose. The argument parsing is done and tested elsewhere in the project. What I'm interested here is testing the inner plumbing with and without the parsed configuration variables present to prevent regressions in the future.